### PR TITLE
ansiweather 1.04 (move from head-only)

### PR DESCRIPTION
**Pull request by jawshooah**
Refs Homebrew/homebrew-head-only#157

### CODE REVIEW

File: Library/Formula/ansiweather.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Ansiweather < Formula
+  homepage "https://github.com/fcambus/ansiweather"
```
Comment (mikemcquaid) [inline Library/Formula/ansiweather.rb]: Please add a `desc` above this line, thanks!

File: Library/Formula/ansiweather.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "jq"
+
+  def install
+    bin.install "ansiweather"
+  end
+
+  test do
+    output = shell_output("#{bin}/ansiweather -l Ithaca,NY")
+    assert_match /Current weather in Ithaca/, output
```
Comment (mikemcquaid) [inline Library/Formula/ansiweather.rb]: Test is failing on 10.10: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/30802/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_ansiweather/

Comment (jawshooah) [inline Library/Formula/ansiweather.rb]: Strange, I can't reproduce locally. Is the system Ruby on the 10.10 VM unable to process Unicode?

Comment (mikemcquaid) [inline Library/Formula/ansiweather.rb]: Perhaps you need to setup an environment variable with `ENV`?

Comment (jawshooah) [inline Library/Formula/ansiweather.rb]: Doubtful, I can't find any references to environment variables in `ansiweather`'s documentation or source code.

Comment (jawshooah) [inline Library/Formula/ansiweather.rb]: I'll try adding a test `.ansiweatherrc` file with unicode characters disabled.

Comment (mikemcquaid) [inline Library/Formula/ansiweather.rb]: good idea!

**Comment by dunn** [comment]
The test is failing:
```
==> /usr/local/Cellar/ansiweather/1.02/bin/ansiweather -l Ithaca,NY -s false
Error: ansiweather: failed
invalid byte sequence in US-ASCII
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/minitest/unit.rb:302:in `=~'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/minitest/unit.rb:302:in `assert_match'
/usr/local/Library/Formula/ansiweather.rb:17:in `block in <class:Ansiweather>'
/usr/local/Library/Homebrew/formula.rb:929:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:16:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:925:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
```
It might just need `ENV["LC_ALL"] = "en_US.UTF-8"` at the beginning of the test block.

**Comment by DomT4** [comment]
Ping.

**Comment by jawshooah** [comment]
Added the fix proposed by @dunn above, we'll see if the test passes on the build machine.

**Comment by mikemcquaid** [comment]
Still failing unfortunately: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32213/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_ansiweather/

**Comment by jawshooah** [comment]
I honestly have no idea what's going wrong here :disappointed:

Feel free to close, I guess. I have no urgent need for this tool, I just noticed that a versioned release was available and figured it should be moved out of the head-only repo.

**Comment by jawshooah** [comment]
@fcambus, do you have any suggestions?

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by fcambus** [comment]
@jawshooah You are setting symbols to false so it should not output any UTF-8 characters. Could the error be caused by the ANSI sequences?

Ans yes, it would be nice to have AnsiWeather in homebrew. Thanks for taking care of that :)

**Comment by fcambus** [comment]
For info, 1.03 has been released today.

**Comment by jawshooah** [comment]
Thanks @fcambus, I've updated the formula to 1.03.

Maybe ansiweather could only output ANSI color codes if it's run in a TTY?

**Comment by fcambus** [comment]
Sounds like a good idea. Just commited support on master branch for a '-a' option allowing to toggle ANSI color codes on and off.

Try : ansiweather -l Ithaca,NY -a false -s false

Does that help regarding the test?

**Comment by jawshooah** [comment]
Cool! Only way to find out is to push another release so I can update the formula and @BrewTestBot can test it.

**Comment by fcambus** [comment]
Just released 1.04, let's see how it fares this time with the '-a' option set to false.

**Comment by fcambus** [comment]
@jawshooah any update on this? Did the ascii only output solve the testing issue?

**Comment by jawshooah** [comment]
@fcambus Just pushed the update, we'll see how it goes.

**Comment by DomT4** [comment]
Failed again. Try [this sort of thing](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/abduco.rb#L22-L23) maybe :confused:.

**Comment by jawshooah** [comment]
@DomT4 That worked! Build failure is just due to the duplication with homebrew-head-only.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/59eaeef181b616fade910c836d3b0aef144107e3. Thank you for sticking with this & your contribution to Homebrew @jawshooah!

**Pull request closed without merging**